frontend/discourse/app/templates/preferences/profile.gjs

```diff
@@ -1,246 +1,396 @@
-      <DButton
-        @icon="globe"
-        @label="user.use_current_timezone"
-        @action={{@controller.useCurrentTimezone}}
-        class="btn-default"
-      />
-    </div>
-
-    {{#if @controller.model.can_change_location}}
-      <div
-        class="control-group pref-location"
-        data-setting-name="user-location"
-      >
-        <label class="control-label" for="edit-location">{{i18n
-            "user.location"
-          }}</label>
-        <div class="controls">
-          <Input
-            @type="text"
-            @value={{@controller.model.location}}
-            class="input-xxlarge"
-            id="edit-location"
-          />
-        </div>
-      </div>
-    {{/if}}
-
-    {{#if @controller.model.can_change_website}}
-      <div class="control-group pref-website" data-setting-name="user-website">
-        <label class="control-label" for="edit-website">{{i18n
-            "user.website"
-          }}</label>
-        <div class="controls">
-          <Input
-            @type="text"
-            @value={{@controller.model.website}}
-            class="input-xxlarge"
-            id="edit-website"
-          />
-        </div>
-      </div>
-    {{/if}}
-  {{/unless}}
-
-  {{#each @controller.userFields as |uf|}}
-    <div class="control-group" data-setting-name="user-user-fields">
-      <UserField @field={{uf.field}} @value={{uf.value}} />
-    </div>
-  {{/each}}
-  <div class="clearfix"></div>
-
-  {{#unless @controller.showEnforcedRequiredFieldsNotice}}
-    {{#if @controller.siteSettings.allow_profile_backgrounds}}
-      {{#if @controller.canUploadProfileHeader}}
-        <div
-          class="control-group pref-profile-bg"
-          data-setting-name="user-profile-bg"
+const validationFor = (field) => {
+  if (!get(field, "required")) {
+    return null;
+  }
+  return get(field, "field_type") === "confirm" ? "accepted" : "required";
+};
+
+class MutableField extends Component {
+  get value() {
+    return this.args.value;
+  }
+
+  set value(val) {
+    this.args.set(val);
+  }
+
+  <template>{{yield this}}</template>
+}
+
+export default class Profile extends Component {
+  @service dialog;
+  @service modal;
+
+  constructor() {
+    super(...arguments);
+  }
+
+  @action
+  updateTimezone(field, timezone) {
+    field.set(timezone);
+  }
+
+  @action
+  useCurrentTimezone(field) {
+    field.set(moment.tz.guess(true));
+  }
+
+  @action
+  profileBackgroundUploadDone(field, upload) {
+    field.set(upload.url);
+  }
+
+  @action
+  cardBackgroundUploadDone(field, upload) {
+    field.set(upload.url);
+  }
+
+  @action
+  setFeaturedTopic(field, v) {
+    field.set(v);
+  }
+
+  @action
+  async showFeaturedTopicModal(field) {
+    await this.modal.show(FeatureTopicOnProfileModal, {
+      model: {
+        user: this.args.controller.model,
+        setFeaturedTopic: (v) => this.setFeaturedTopic(field, v),
+      },
+    });
+    document.querySelector(".feature-topic-on-profile-btn")?.focus();
```

Comment:
Why is this needed?

Reply:
This is just mimicking what already happens on the function in the controller. It is for A11Y reasons, once completing the action triggered by the button, to return the focus to the button, so users could continue tabbing through items.